lib/api/v3/work_packages/work_package_representer.rb

```diff
@@ -301,13 +309,18 @@ def activities
         end
 
         def watchers
-          watchers =
-            represented.watcher_users.order(User::USER_FORMATS_STRUCTURE[Setting.user_format])
-          watchers.map do |watcher|
-            ::API::V3::Users::UserRepresenter.new(watcher,
-                                                  work_package: represented,
-                                                  current_user: current_user)
-          end
+          # TODO/LEGACY: why do we need to ensure a specific order here?
+          watchers = represented.watcher_users.order(User::USER_FORMATS_STRUCTURE[Setting.user_format])
+          total = watchers.count
+          self_link = api_v3_paths.work_package_watchers(represented.id)
+
+          # FIXME/LEGACY: we pass the WP as context?!? that makes a difference!!!
+          # tl;dr: the embedded user representer must not be better than any other user representer
```

Comment:
The user representer actually makes use of the pased work package and provides more links in that case (watcher specific links).

This is in contrast to everything we discussed today about separating between user and watcher. However, I found it out of scope of this (already grown) PR to also address this issue, because at least our angular client relies on those fields ATM.